### PR TITLE
Removed orphaned owners

### PR DESCRIPTION
**Pull request by kevinlinxc**
Trying to address [https://app.honeybadger.io/projects/40972/faults/68202479](https://app.honeybadger.io/projects/40972/faults/68202479), e.g. https://rubygems.org/gems/rmagick4j/owners not showing anything because the user got deleted.

This PR is a migration to delete ownership instances with nil users, and to make user deletion delete related ownerships in the future. My first time writing a migration, so I'm unsure if this is correct.

### CODE REVIEW

**Comment by kevinlinxc** [comment]
It was recommended that I split this into two PRs, the deletion first and then the foreign key removal. This is so they could be rolled back separately.
This PR will be for the deletion, since that has to happen first.

File: db/migrate/20220609221942_remove_orphaned_owners.rb
Comment (jchestershopify) [inline db/migrate/20220609221942_remove_orphaned_owners.rb:2]: I think we should still use `up` and `down` and [raise `ActiveRecord::IrreversibleMigration` from `down`](https://guides.rubyonrails.org/active_record_migrations.html#using-the-up-down-methods). Rails can't deduce how to undelete a record.

File: db/migrate/20220609221942_remove_orphaned_owners.rb
Comment (jchestershopify) [inline db/migrate/20220609221942_remove_orphaned_owners.rb:4]: What happened to adding a foreign key?

Comment (jchestershopify) [reply db/migrate/20220609221942_remove_orphaned_owners.rb:4]: Ah, I see from [this conversation in Slack](https://shopify.slack.com/archives/GM1850REJ/p1655242779870469) that the intention is to split the migration into two individual migrations.

File: db/migrate/20220609221942_remove_orphaned_owners.rb
Comment (kevinlinxc) [reply db/migrate/20220609221942_remove_orphaned_owners.rb]: Implemented this

**Comment by kevinlinxc** [comment]
My testing:
1. Run elasticsearch, memcached, postgresql
2. Seed database with gems and owners https://gist.github.com/kevinlinxc/82044f2674fbc4faf2c517e825b39818
3. Find an owner, gem, and ownership, e.g. parser gem (id 74822) with owner squid@games.com (id 7), Ownership id 77
4. Create an account, set email verified to true in console, and add myself as owner for that gem

`User.where(email: "kevin.lin@shopify.com").first.update_attribute(:email_confirmed, true)`

`Ownership.create_confirmed(Rubygem.where(id: 74822).first, User.where(email: "kevin.lin@shopify.com").first, Us
er.where(id: 7))`

5. Visit gems/parser/owners to ensure it works normally:
image
6. Delete the other owner

`User.where(email: "squid@games.com").delete_all`
As expected, the user is gone but their ownerships persist: 
image
Reloading the /owners page yields the error:
image

7. Run the migration to remove ownerships:

image

Website looks normal now:

image

**Review by jenshenny — APPROVED** [review]
Tested and it looks ok to me!

**Review by simi — APPROVED** [review]
LGTM! But I would like to get @sonalkr132 approval before merge in here since it is DB related.

**Comment by sonalkr132** [comment]
Please dry run this.

File: db/migrate/20220609221942_remove_orphaned_owners.rb
Comment (simi) [inline db/migrate/20220609221942_remove_orphaned_owners.rb:3]: :information_source: This can be written today as `Ownership.where.missing(:user).destroy_all`.

Comment (jenshenny) [reply db/migrate/20220609221942_remove_orphaned_owners.rb]: Oh nice, updated!

**Comment by jchestershopify** [comment]
Should this and #3101 be merged?

**Comment by simi** [comment]
I'm awaiting @sonalkr132 approval. Once done, I can deploy this and also following change (one by one).

**Pull request merged**
